src/auth/AxiosJwtAuthService.js

```diff
@@ -0,0 +1,303 @@
+      unauthorizedError.isRedirecting = true;
+      throw unauthorizedError;
+    }
+
+    return this.getAuthenticatedUser();
+  }
+
+  /**
+   * Fetches additional user account information for the authenticated user and merges it into the
+   * existing authenticatedUser object, available via getAuthenticatedUser().
+   *
+   * ```
+   *  console.log(authenticatedUser); // Will be sparse and only contain basic information.
+   *  await hydrateAuthenticatedUser()
+   *  const authenticatedUser = getAuthenticatedUser();
+   *  console.log(authenticatedUser); // Will contain additional user information
+   * ```
+   *
+   * @returns {Promise<null>}
+   */
+  async hydrateAuthenticatedUser() {
+    const user = this.getAuthenticatedUser();
+    if (user !== null) {
+      const response = await this.authenticatedHttpClient
+        .get(`${this.config.LMS_BASE_URL}/api/user/v1/accounts/${user.username}`);
+      this.setAuthenticatedUser({ ...user, ...camelCaseObject(response.data) });
+    }
+  }
+
+  /**
+ * Adds authentication defaults and interceptors to an HTTP client instance.
+ *
+ * @param {HttpClient} newHttpClient
+ * @param {Object} config
+ * @param {string} [config.REFRESH_ACCESS_TOKEN_ENDPOINT]
+ * @param {string} [config.ACCESS_TOKEN_COOKIE_NAME]
+ * @param {string} [config.CSRF_TOKEN_API_PATH]
+ * @returns {HttpClient} A configured Axios HTTP client.
+ */
+  addAuthenticationToHttpClient(newHttpClient) {
+    const httpClient = Object.create(newHttpClient);
+    // Set withCredentials to true. Enables cross-site Access-Control requests
+    // to be made using cookies, authorization headers or TLS client
+    // certificates. More on MDN:
+    // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
+    httpClient.defaults.withCredentials = true;
+
+    // Axios interceptors
+
+    // The JWT access token interceptor attempts to refresh the user's jwt token
+    // before any request unless the isPublic flag is set on the request config.
+    const refreshAccessTokenInterceptor = createJwtTokenProviderInterceptor({
+      jwtTokenService: this.jwtTokenService,
+      shouldSkip: axiosRequestConfig => axiosRequestConfig.isPublic,
+    });
+    // The CSRF token intercepter fetches and caches a csrf token for any post,
+    // put, patch, or delete request. That token is then added to the request
+    // headers.
+    const attachCsrfTokenInterceptor = createCsrfTokenProviderInterceptor({
+      csrfTokenService: this.csrfTokenService,
+      CSRF_TOKEN_API_PATH: this.config.CSRF_TOKEN_API_PATH,
+      shouldSkip: (axiosRequestConfig) => {
+        const { method, isCsrfExempt } = axiosRequestConfig;
+        const CSRF_PROTECTED_METHODS = ['post', 'put', 'patch', 'delete'];
+        return isCsrfExempt || !CSRF_PROTECTED_METHODS.includes(method);
+      },
+    });
+
+    const processAxiosRequestErrorInterceptor = createProcessAxiosRequestErrorInterceptor({
```

Comment:
This is a change - this interceptor didn't have a `create*` function prior to this.